GameManager.gd

```diff
@@ -33,15 +34,44 @@ func add_points(value):
 func _ready():
 	print("game manage ready")
 
-
 func reset():
 	overworld_entrance_position = null
 	level = 0
 	points = 0
 	
-func createSeed():
+	if has_node("/root/world"): # Game is in progress.
+		get_node("/root/world").queue_free()
+	get_node("/root/players").queue_free()
+	
+func create_seed():
 	if (initial_seed == 0):
 		randomize()
 		initial_seed = randi()
 	print("Initial Game Seed:"+str(initial_seed))
 	seed(initial_seed)
+	
+remote func get_seed():
+	GameManager.initial_seed
+	
+remote func set_seed(new_seed):
+	initial_seed = new_seed
+
+func change_scene(scene_resource, player = null):
+	# load new scene
+	var new_world = load(scene_resource).instance()
+	new_world.name = "world"
+	var root = get_node("/root")
+	
+	# clean up old one
+	if (has_node("/root/world")):
+		var prev_world = get_node("/root/world")
+		prev_world.name = "previous_world"
+		prev_world.queue_free()
+	
+	root.add_child(new_world)
+	
+	# fix players index
+	if (has_node("/root/players")):
+		root.move_child(get_node("/root/players"), root.get_child_count() -1)
```

Comment:
I'm curious how the players are showing up in different dungeons if this is the strategy we are using.  In theory both players should be on both dungeons since they are together in a single node unless you've done something to hide/disable the other players puppet when they aren't together.